Common++/header/IpAddresses.h

```diff
@@ -0,0 +1,551 @@
+
+	private:
+		uint8_t m_Bytes[16];
+	}; // class IPv6Address
+
+
+	// Creation of IPv6Address
+
+	/**
+	 * A function that creates an instance of the class IPv6Address out of 16-byte array.
+	 * @param[in] bytes The address as 16-byte array in network byte order
+	 * @return An instance of class IPv6Address
+	 */
+	inline IPv6Address makeIPv6Address(const uint8_t bytes[16]) { return IPv6Address(bytes); }
+
+	/**
+	 * A function that creates an instance of the class out of string (char*) value
+	 * If the string doesn't represent a valid IPv6 address an errorCode will be set to non-zero value
+	 * @param[in] addressAsString The string (char*) representation of the address
+	 * @param[out] errorCode Contains 0 if a string represents a valid address, otherwise a non-zero value
+	 * @return An instance of class IPv6Address. If an error occured the return address will be an unspecified/zero
+	 */
+	IPv6Address makeIPv6Address(const char* addrAsString, int& errorCode);
+
+	/**
+	 * A function that creates an instance of the class out of std::string value
+	 * If the string doesn't represent a valid IPv6 address an errorCode will be set to non-zero value
+	 * @param[in] addressAsString The std::string representation of the address
+	 * @param[out] errorCode Contains 0 if a string represents a valid address, otherwise a non-zero value
+	 * @return An instance of class IPv6Address. If an error occured the return address will be an unspecified/zero
+	 */
+	inline IPv6Address makeIPv6Address(const std::string& addrAsString, int& errorCode) {	return makeIPv6Address(addrAsString.c_str(), errorCode); }
+
+
+
+	///**
+	// * @class IPAddress
+	// * The class is a version-independent representation for an IP address
+	// */
+	class IPAddress
+	{
+	public:
+		/**
+		 * An enum representing the address type: IPv4 or IPv6
+		 */
+		enum AddressType
+		{
+			/**
+			 * IPv4 address type
+			 */
+			IPv4AddressType,
+			/**
+			 * IPv6 address type
+			 */
+			IPv6AddressType
+		};
+
+		/**
+		 * A default constructor that creates an instance of the class with unspecified IPv4 address
+		 */
+		IPAddress() : m_Type(IPv4AddressType) {}
+
+		/**
+		 * A constructor that creates an instance of the class out of IPv4Address.
+		 * @param[in] addr A const reference to instance of IPv4Address
+		 */
+		IPAddress(const IPv4Address& addr) : m_Type(IPv4AddressType), m_IPv4(addr) {}
+
+		/**
+		 * A constructor that creates an instance of the class out of IPv6Address.
+		 * @param[in] addr A const reference to instance of IPv6Address
+		 */
+		IPAddress(const IPv6Address& addr) : m_Type(IPv6AddressType), m_IPv6(addr) {}
+
+		/**
+		 * Overload of an assignment operator.
+		 * @param[in] addr A const reference to instance of IPv4Address
+		 */
+		inline IPAddress& operator=(const IPv4Address& addr);
+
+		/**
+		 * Overload of an assignment operator.
+		 * @param[in] addr A const reference to instance of IPv6Address
+		 */
+		inline IPAddress& operator=(const IPv6Address& addr);
```

Comment:
should we add another overload of this method that takes `IPAddress` as a parameter?

Reply:
No we should not. The compiler will generate a default assignment operator.